src/IslandoraUtils.php

```diff
@@ -279,12 +279,17 @@ static function (string $field) {
       $unionQuery->union($queryPart);
     }
 
+    // Mariadb optimization:
+    // do not correlate the subquery with the base table.
+    $nonCorrelatedSubquery = $this->database->select($unionQuery, 'sq');
+    $nonCorrelatedSubquery->fields('sq');
```

Comment:
Just to give a nod, and potentially useful reference for the future:

```suggestion
    // Mariadb optimization: Explicitly avoid execution as a correlated subquery.
    // Adapted from: https://stackoverflow.com/a/6157797
    $nonCorrelatedSubquery = $this->database->select($unionQuery, 'sq');
    $nonCorrelatedSubquery->fields('sq');
```